### PR TITLE
Fix: Make ⌘A's select all the blocks again

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
## Description
Fixes: https://github.com/WordPress/gutenberg/issues/7711
Regressed in: https://github.com/WordPress/gutenberg/pull/7644

Moves the logic to handle meta+a to be before the check for event.nativeEvent.defaultPrevented. The event was default prevented by TinyMCE.

## How has this been tested?
Some smoke testing.
Added multiple paragraphs with text and checked that pressing ⌘A's multiple selected all the blocks.

### CODE REVIEW

File: editor/components/writing-flow/index.js [hunk 1]
```diff
@@ -190,7 +190,7 @@ class WritingFlow extends Component {
 
 		// Aobrt if navigation has already been handled (e.g. TinyMCE inline
```
Comment (mtias) [inline editor/components/writing-flow/index.js]: There's a typo here "Aobrt"

**Comment by aduth** [comment]
End-to-end test failure is legitimate. This is basically undoing what was meant to be fixed by #7644

**Review by aduth — DISMISSED** [review]
Re-regresses #7644 . `_navigationHandled` isn't a thing. It's basically the same as if the condition isn't there at all.

**Comment by afercia** [comment]
Also related: https://github.com/WordPress/gutenberg/issues/7445

**Comment by jorgefilipecosta** [comment]
This PR was updated it now reorders the check for ⌘A event to be before the check for event.nativeEvent.defaultPrevented. It seems TinyMCE applies a prevent default in the event.

**Comment by aduth** [comment]
Future tasks:

- Split `WritingFlow#onKeyDown` into more sensible isolated function handlers (we shouldn't have both block traversal and Cmd+A behaviors in the same function, it contributed to the introduction of the regression)
- Try to interoperate better with TinyMCE's `preventDefault`, e.g. perhaps exploring the [`SelectAll` command](https://github.com/tinymce/tinymce/blob/0f7a0f12667bde6eae9377b50b797f4479aa1ac7/src/core/main/ts/util/Quirks.ts#L196-L203) which is the source of the `preventDefault`, either stripping the default TinyMCE behavior, refactoring our 2x Cmd+A to act in response to the `SelectAll` command (similar to Undo/Redo propagation), or abandoning effort to use native `preventDefault` in favor of an implementation-specific property like which had existed prior to #7644

On glance, this looks good. Will take a closer look soon. Travis appears to be having a bad day.

File: editor/components/writing-flow/index.js [hunk 2]
```diff
@@ -204,15 +198,8 @@ class WritingFlow extends Component {
...
-
 		const isNavEdge = isVertical ? isVerticalEdge : isHorizontalEdge;
 
-		if ( ! isVertical ) {
-			this.verticalRect = null;
-		} else if ( ! this.verticalRect ) {
-			this.verticalRect = computeCaretRect( target );
-		}
-
 		if ( ! isNav ) {
```
Comment (iseulde) [inline editor/components/writing-flow/index.js]: Worth leaving a comment on the order?

Comment (jorgefilipecosta) [reply editor/components/writing-flow/index.js]: Well remembered, the comment was added.

**Review by ellatrix — APPROVED** [review]
Looks fine to me. Added an e2e test to ensure we don't break it in the future. Arrow nav still works as expected.

**Comment by jorgefilipecosta** [comment]
Thank you for the review and for suggesting this approach @aduth. Thank you @iseulde for the review and for the end to end test 👍

**Comment by aduth** [comment]
Thanks for the added end-to-end test!

**Comment by mcsf** [comment]
Just noting that, at least as of today:
- Firefox requires three <kbd>⌘A</kbd> presses.
- Safari doesn't seem to care about this feature at all.

**Comment by mcsf** [comment]
I see the above is sort of tracked in #8180, so disregard. :)

**Pull request merged**